Makefile

```diff
@@ -38,15 +54,15 @@ default: netplan/_features.py generate netplan-dbus dbus/io.netplan.Netplan.serv
 %.o: src/%.c
 	$(CC) $(BUILDFLAGS) $(CFLAGS) $(LDFLAGS) -c $^ `pkg-config --cflags --libs glib-2.0 gio-2.0 yaml-0.1 uuid`
 
-libnetplan.so.$(NETPLAN_SOVER): parse.o netdef.o netplan.o util.o validation.o error.o parse-nm.o names.o
-	$(CC) -shared -Wl,-soname,libnetplan.so.$(NETPLAN_SOVER) $(BUILDFLAGS) $(CFLAGS) $(LDFLAGS) -o $@ $^ `pkg-config --libs glib-2.0 gio-2.0 yaml-0.1`
+libnetplan.so.$(NETPLAN_SOVER): $(SRCS)
+	$(CC) -shared -Wl,-soname,libnetplan.so.$(NETPLAN_SOVER) $(BUILDFLAGS) $(CFLAGS) `pkg-config --cflags glib-2.0 gio-2.0 yaml-0.1 uuid` -o $@ $^ `pkg-config --libs glib-2.0 gio-2.0 yaml-0.1 uuid` -fvisibility=hidden
```

Comment:
Indeed. However, it's usually easier to move CFLAGS around than LDFLAGS, as where the libraries appear affects the symbol resolution. Your patch makes the tests fail because they can't find some GLib symbols for some reason... A fixed version will be shipped in V2.